### ISSUE TITLE
TERRAFORM CRASH (version 1.6.0) on MAC ARM

### ISSUE THREAD

**Issue by yebolenko** [hashicorp/terraform#34020]
### Terraform Version

```shell
1.6.0
```

### Terraform Configuration Files

Not Applicable

### Debug Output

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

checkable object status report for unexpected checkable object module.merged_eks_definitions.var.maps
goroutine 53480 [running]:
runtime/debug.Stack()
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:16 +0x1c
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        /Users/runner/work/terraform/terraform/internal/logging/panic.go:58 +0x164
panic({0x10302a8a0?, 0x14009a40810?})
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:920 +0x26c
github.com/hashicorp/terraform/internal/checks.(*State).reportCheckResult(0x140048be150, {0x10365c8a8, 0x1400856aae0}, 0x0?, 0x0, 0x50)
        /Users/runner/work/terraform/terraform/internal/checks/state_report.go:106 +0x2a8
github.com/hashicorp/terraform/internal/checks.(*State).ReportCheckResult(0x103466720?, {0x10365c8a8?, 0x1400856aae0?}, 0x14009617678?, 0x1?, 0x2063e98?)
        /Users/runner/work/terraform/terraform/internal/checks/state_report.go:71 +0xd4
github.com/hashicorp/terraform/internal/terraform.evalVariableValidations({{0x14008086b40, 0x1, 0x1}, {{}, {0x14002063e98, 0x4}}}, 0x14002fa6b60, {0x10c8d1018, 0x14000a28640}, {0x103672450, ...})
        /Users/runner/work/terraform/terraform/internal/terraform/eval_variable.go:252 +0x85c
github.com/hashicorp/terraform/internal/terraform.(*nodeModuleVariable).Execute(0x14000f41f80, {0x103672450, 0x140039e57a0}, 0x0?)
        /Users/runner/work/terraform/terraform/internal/terraform/node_module_variable.go:198 +0x2d4
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0x14007d218c0, {0x103672450, 0x140039e57a0}, {0x10c76bac8, 0x14000f41f80})
        /Users/runner/work/terraform/terraform/internal/terraform/graph_walk_context.go:143 +0xa0
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1({0x1033a2a60, 0x14000f41f80})
        /Users/runner/work/terraform/terraform/internal/terraform/graph.go:78 +0x280
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0x14002c1e180, {0x1033a2a60, 0x14000f41f80}, 0x140080dbc00)
        /Users/runner/work/terraform/terraform/internal/dag/walk.go:384 +0x2a8
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update in goroutine 52014
        /Users/runner/work/terraform/terraform/internal/dag/walk.go:307 +0xb0c

### Expected Behavior

CRASH

### Actual Behavior

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

checkable object status report for unexpected checkable object module.merged_eks_definitions.var.maps
goroutine 53480 [running]:
runtime/debug.Stack()
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:16 +0x1c
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        /Users/runner/work/terraform/terraform/internal/logging/panic.go:58 +0x164
panic({0x10302a8a0?, 0x14009a40810?})
        /Users/runner/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:920 +0x26c
github.com/hashicorp/terraform/internal/checks.(*State).reportCheckResult(0x140048be150, {0x10365c8a8, 0x1400856aae0}, 0x0?, 0x0, 0x50)
        /Users/runner/work/terraform/terraform/internal/checks/state_report.go:106 +0x2a8
github.com/hashicorp/terraform/internal/checks.(*State).ReportCheckResult(0x103466720?, {0x10365c8a8?, 0x1400856aae0?}, 0x14009617678?, 0x1?, 0x2063e98?)
        /Users/runner/work/terraform/terraform/internal/checks/state_report.go:71 +0xd4
github.com/hashicorp/terraform/internal/terraform.evalVariableValidations({{0x14008086b40, 0x1, 0x1}, {{}, {0x14002063e98, 0x4}}}, 0x14002fa6b60, {0x10c8d1018, 0x14000a28640}, {0x103672450, ...})
        /Users/runner/work/terraform/terraform/internal/terraform/eval_variable.go:252 +0x85c
github.com/hashicorp/terraform/internal/terraform.(*nodeModuleVariable).Execute(0x14000f41f80, {0x103672450, 0x140039e57a0}, 0x0?)
        /Users/runner/work/terraform/terraform/internal/terraform/node_module_variable.go:198 +0x2d4
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0x14007d218c0, {0x103672450, 0x140039e57a0}, {0x10c76bac8, 0x14000f41f80})
        /Users/runner/work/terraform/terraform/internal/terraform/graph_walk_context.go:143 +0xa0
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1({0x1033a2a60, 0x14000f41f80})
        /Users/runner/work/terraform/terraform/internal/terraform/graph.go:78 +0x280
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0x14002c1e180, {0x1033a2a60, 0x14000f41f80}, 0x140080dbc00)
        /Users/runner/work/terraform/terraform/internal/dag/walk.go:384 +0x2a8
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update in goroutine 52014
        /Users/runner/work/terraform/terraform/internal/dag/walk.go:307 +0xb0c

### Steps to Reproduce

terraform init
terraform apply

**Comment by qbvysakh** [comment]
I am also facing the same issue when running from the GitHub actions pipeline. But it is working fine from local terraform installation. The code was working fine till a few days ago.
For me, the issue is causing the [terraform-aws-rds](https://github.com/terraform-aws-modules/terraform-aws-rds) module. 

```value is marked, so must be unmarked first
goroutine 3101 [running]:
runtime/debug.Stack()
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:24 +0x5e
runtime/debug.PrintStack()
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/debug/stack.go:16 +0x13
github.com/hashicorp/terraform/internal/logging.PanicHandler()
	/home/runner/work/terraform/terraform/internal/logging/panic.go:58 +0x13b
panic({0x2b92ec0?, 0x38c4c30?})
	/opt/hostedtoolcache/go/1.21.1/x64/src/runtime/panic.go:920 +0x270
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
	/home/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.14.0/cty/marks.go:141
github.com/zclconf/go-cty/cty.Value.AsString({{{0x3902720?, 0xc000142bd1?}}, {0x2e1de40?, 0xc0024a2600?}})
	/home/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.14.0/cty/value_ops.go:1385 +0x47
github.com/zclconf/go-cty/cty.Value.Range({{{0x3902720?, 0xc000142bd1?}}, {0x2e1de40?, 0xc0024a2600?}})
	/home/runner/go/pkg/mod/github.com/zclconf/go-cty@v1.14.0/cty/value_range.go:53 +0x285
github.com/hashicorp/hcl/v2/hclsyntax.(*ConditionalExpr).Value(0xc000892f50, 0xc0024a25e8)
	/home/runner/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.18.0/hclsyntax/expression.go:746 +0xd6a
github.com/hashicorp/terraform/internal/lang.(*Scope).EvalExpr(0xc00483a750, {0x3902df8?, 0xc000892f50}, {{0x3902f48?, 0x5130520?}})
	/home/runner/work/terraform/terraform/internal/lang/eval.go:175 +0x1bd
github.com/hashicorp/terraform/internal/terraform.(*nodeModuleVariable).evalModuleVariable(0xc000a1f3e0, {0x391fb60?, 0xc001e2c700?}, 0x80?)
	/home/runner/work/terraform/terraform/internal/terraform/node_module_variable.go:248 +0x225
github.com/hashicorp/terraform/internal/terraform.(*nodeModuleVariable).Execute(0xc000a1f3e0, {0x391fb60, 0xc001e2c700}, 0x4)
	/home/runner/work/terraform/terraform/internal/terraform/node_module_variable.go:183 +0x12c
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc002e127e0, {0x391fb60, 0xc001e2c700}, {0x7f7df815a980, 0xc000a1f3e0})
	/home/runner/work/terraform/terraform/internal/terraform/graph_walk_context.go:143 +0xbe
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1({0x2f07ca0, 0xc000a1f3e0})
	/home/runner/work/terraform/terraform/internal/terraform/graph.go:78 +0x375
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc000a1f4a0, {0x2f07ca0, 0xc000a1f3e0}, 0xc004053ec0)
	/home/runner/work/terraform/terraform/internal/dag/walk.go:384 +0x2e5
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update in goroutine 1304
	/home/runner/work/terraform/terraform/internal/dag/walk.go:307 +0xde8```

**Comment by qbvysakh** [comment]
Using an older version of terraform(1.5.7) fixed the issue for me

**Comment by liamcervante** [comment]
Hi @yebolenko, thanks for filing this. I believe this is a duplicate of #34014, so I will be closing this as a duplicate and you can follow the resolution in the linked issue.

@qbvysakh, I think your crash is slightly different but a duplicate of #33977 which will have a fix released in the next path release for the v1.6 series.

Thanks!
